### PR TITLE
VulkanSceneGraph 1.0.0 ,1.0.3, 1.0.5

### PR DESCRIPTION
**Pull request by psyinf**
Specify library name and version:  vsg 1.0.0

VulkanSceneGraph  is the successor to OpenSceneGraph developed by Robert Osfield. I'd like to enable broader access to it by making it available via conan. The current version of the conan package is rather minimal and due to the nature of the VSG project, other packages are likely to be needed to have a productive VSG environment. 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/vsg/all/conanfile.py
Comment (franramirez688) [inline recipes/vsg/all/conanfile.py:4]:
```suggestion
from conan.tools.files import get, copy, rm, rmdir, collect_libs
```

File: recipes/vsg/all/conanfile.py
Comment (franramirez688) [inline recipes/vsg/all/conanfile.py:8]:
```suggestion
```

File: recipes/vsg/all/conanfile.py
Comment (franramirez688) [inline recipes/vsg/all/conanfile.py:118]:
```suggestion
        self.cpp_info.libs = collect_libs(self)
```

File: recipes/vsg/all/conanfile.py
Comment (franramirez688) [inline recipes/vsg/all/conanfile.py:95]:
```suggestion
```

**Review by franramirez688 — COMMENTED** [review]
Thanks for this contribution!
LGTM! Let's apply those suggestions to see if it also passes for Conan v2 😁

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution. I only have a minor request :)

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:21]: Hi @RubenRBS, thanks for the suggestions: The suggested addition is addressed here 2ebb28c56

**Review by franramirez688 — APPROVED** [review]
👏

**Review by RubenRBS — CHANGES_REQUESTED** [review]
More reviews! Sorry I didnt catch them all in the first round

File: recipes/vsg/all/conanfile.py
Comment (jcar87) [inline recipes/vsg/all/conanfile.py:54]:
```suggestion
        self.requires("vulkan-loader/1.3.239.0", transitive_headers=True)
```

Comment (jcar87) [reply recipes/vsg/all/conanfile.py:54]: Applying this suggestion as `1.3.239.0` is the most recent version we deliver in Conan 2.0 - unless there's any specific reasons to stay on 204??

Comment (psyinf) [reply recipes/vsg/all/conanfile.py]: No, there is no particular reason for the .204. It simply was the version referenced by vsg 1.0.0 IIRC.

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:76]: What do you mean by "add the commit to be checked out to the conandata.yaml"? I found no way to check out a specific version of a "dependency" at source level with a specific version here. Can you elaborate on this?

Comment (psyinf) [reply recipes/vsg/all/conanfile.py:76]: The best I came up with is to add a new section for the submodule url and revision per version to the conandata.yml and retrive it like so:
```
self.run("git clone " + self.conan_data["submodules"][self.version]["url"]  + " src/glslang")
self.run("cd src/glslang && git reset --hard " + self.conan_data["submodules"][self.version]["rev"])
```

Comment (psyinf) [reply recipes/vsg/all/conanfile.py]: I tried to add the "submodule" information to the conandata but this didn't work out as planed - I still need to be able to check out a specific branch of a repository to a sub-directory of the source. Can this even been done via the conandata.yml?

Comment (RubenRBS) [reply recipes/vsg/all/conanfile.py]: I expected to be able to add custom keys under the `sources` key in that file 🤔 If that does not work, I'll need to check how to best do it

Comment (psyinf) [reply recipes/vsg/all/conanfile.py]: That unfortunately doesn't work either (but would sort of solve the issue). Currently it looks like the author of VSG moved to use one branch per major/minor combination. I'm quite curious if this is a common requirement or if VSG does something exotic here.

**Review by franramirez688 — APPROVED** [review]

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py]: @RubenRBS Have you had the time to find a better solution for this? I changed the code to use the scm-tools but cannot improve on the whereabouts of the version-data

File: recipes/vsg/all/conanfile.py
Comment (RubenRBS) [inline recipes/vsg/all/conanfile.py:78]:
```suggestion
        if self.version in ["1.0.3"]:
            git = Git(self)
            glslang_url = "https://github.com/vsg-dev/glslang.git"
            glslang_branch = "VSG-1.0.x"
            clone_args = ['--depth', '1', '--branch', glslang_branch]
            git.clone(url= glslang_url, args=clone_args, target ="src/glslang")
```

File: recipes/vsg/all/conanfile.py
Comment (RubenRBS) [reply recipes/vsg/all/conanfile.py]: I'm going to ask @uilianries in the office tomorrow, sorry about the delay :)

**Review by RubenRBS — CHANGES_REQUESTED** [review]
New approach for GLS, sorry to give you more work to do, we really appreciate your contributions :)

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:78]: I cannot change how VSG is structured internally (it uses Cmake to pull the branch /formerly a submodule into its source tree). As you can see this is a fork of glslang and it is cloned into the sources of the VSG project. So I don't really understand how to change this to a dependency/requirement without potentially patching a lot of the VSG code.

If patching and using glslang as requirement is the only road here I'm afraid I won't be able to put the necessary effort into  this anytime soon. Apart from this, all other vsg-libraries use exactly the same approach. 
I'll take a dive into this option later this week, but I have the feeling that this might really complicate things in terms for  maintaining.

Do have any hints here? 

P.S. Thanks for your efforts

Comment (RubenRBS) [reply recipes/vsg/all/conanfile.py:78]: Hi @psyinf thanks for the detailed response, it helps a lot! Let me talk this with the team to find a nice way forward :)

Comment (psyinf) [reply recipes/vsg/all/conanfile.py:78]: How do we want to proceed with this? I can try to patch the vsg-cmake in a way it receives the original glslang dependencies for now. This will work for now, but won't help once vsg decides to depend on a custom fork. And as mentioned it might cause trouble later on in vsgImGui etc.

Comment (psyinf) [reply recipes/vsg/all/conanfile.py:78]: @RubenRBS I've tried to patch it by using glslang as a dependency and failed. (The conan package for glslang doesn't seem to export the public headers used by VSG). So for now I'm not really sure if this is solvable at all without either 
a) dropping the support for shader compiling (I guess not really an option for the broader audience)
b) changing VSG itself (I had a hard time to get the author to recognize conan as a multiplicator for the library)

Comment (uilianries) [reply recipes/vsg/all/conanfile.py:78]: The best approach would be creating a separated package for glslang, then, add it as build requirement (is that a build requirement only, right?). In Conan Center there is a package for glslang already available: https://conan.io/center/glslang, but it provided by Khronos group, I don't know how compatible is it, but you could create another package in case they are incompatible, like `vsg-glslang`.

Comment (psyinf) [reply recipes/vsg/all/conanfile.py:78]: @uilianries  Thx for the insight. It is an optional build requirement (but without it, the library is practically severely hampered). The glsl-package on conan-center unfortunately is not an option as the vsg uses some parts that are not exported here. I'll consider the approach you presented by creating a vsg-glslang.
Can we for now simply cut this part out to finalize this PR? If this would be acceptable, I could remove the dependency for now, albeit only a temporary solution (I could make another attempt on integrating the glslang using your proposal later, when I find the time).

Comment (uilianries) [reply recipes/vsg/all/conanfile.py:78]: @psyinf Yes, you can remove  glslang as dependency right now, and in a next PR, when `vsg-glslang` be available, you can add it as build requirement, no problem at all.

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py]: I removed the git operations for now and will try to tackle this by providing a custom vsg-glslang in the near future

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py]: @uilianries @RubenRBS  I've removed the glslang dependency handling for now and will follow your advice, hopefully in the near future. Please re-review and approve if you see fit

Comment (uilianries) [reply recipes/vsg/all/conanfile.py]: @psyinf Thank you! Let's wait for the CI going back to normal, then we can review it again, but it's looking good.

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by franramirez688 — APPROVED** [review]

**Comment by SSE4** [comment]
@RubenRBS can you unblock review, as your question was addressed already?

**Review by franramirez688 — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]
Thanks a lot! I've removed the now unecessary v1 test and simplified the test package a bit, other than that this looks good :)

File: recipes/vsg/all/conanfile.py
Comment (SpaceIm) [inline recipes/vsg/all/conanfile.py:14]: Would it be possible to provide a more meaningful description please?

File: recipes/vsg/all/conanfile.py
Comment (SpaceIm) [inline recipes/vsg/all/conanfile.py:99]: I guess it's a copy/paste from another recipe. Why would there be `.la` files in a CMake build?

File: recipes/vsg/all/conanfile.py
Comment (SpaceIm) [inline recipes/vsg/all/conanfile.py:103]: why not just remove this cmake folder instead of these 2 lines? It doesn't seem to be used or exposed in package_info, so it's useless.

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:14]: I've modified the PR [here](https://github.com/conan-io/conan-center-index/pull/18996)

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:99]: Good catch, refactored in this [PR](https://github.com/conan-io/conan-center-index/pull/18996)

File: recipes/vsg/all/conanfile.py
Comment (psyinf) [reply recipes/vsg/all/conanfile.py:103]: AFAIR other VSG packages such as vsgXchange need the vsg cmake package info. Seems I've hacked around this then in this [PR](https://github.com/conan-io/conan-center-index/pull/18421). Maybe I need some guidance here on how to use the cmake files generated to be used in subsequent packages. ;-)

Comment (SpaceIm) [reply recipes/vsg/all/conanfile.py:103]: It depends on their official usage. Are they already included in official upstream CMake config file of just exposed in installation layout?
For the later you just have to fill cpp_info.buildirs with folder where these cmake files live. For the former, you have to provide path to each file in cmake_build_modules property.

Comment (psyinf) [reply recipes/vsg/all/conanfile.py:103]: you are correct. Can you point me to another recipe that shows how to tackle this correctly?

Comment (SpaceIm) [reply recipes/vsg/all/conanfile.py:103]: cern-root for example

**Pull request merged**
